Applications/WebPortal/WebApp/library/source/pages/account_page/profile_page/profile_page.tsx

```diff
@@ -155,13 +190,68 @@ export class ProfilePage extends React.Component<Properties> {
       if(this.props.displaySize === DisplaySize.SMALL) {
         if(this.props.hasError) {
           return ProfilePage.STYLE.errorMessage;
-        } else {
+        } else if(this.props.submitStatus) {
           return ProfilePage.STYLE.statusMessage;
+        } else {
+          return ProfilePage.STYLE.hidden;
         }
       } else {
         return ProfilePage.STYLE.hidden;
       }
     })();
+    const changePasswordBox = (() => {
+      let passwordButtonEnabled;
+      if(this.state.password1 !== ''
+          && this.state.password2 !== ''
+          && this.props.isPasswordSubmitEnabled) {
+        passwordButtonEnabled = true;
+      } else {
+        passwordButtonEnabled = false;
+      }
+      let status;
+      if(this.state.isPasswordChanged) {
+        status = '';
+      } else if(this.state.localPasswordMessage !== '') {
+        status = this.state.localPasswordMessage;
+      } else {
+        status = this.props.submitPasswordStatus;
+      }
+      if(this.props.hasPassword) {
+        return (
```

Comment:
You're returning ```null``` if there's no password. To avoid wasting all the work you previously did, perform this check at the beginning of the function and return ```null``` right away. If there is a password only then do you perform all this work and return a component.